### PR TITLE
feat(secrets): Secret Management [CRUD]

### PR DESCRIPTION
**Pull request by yasinmiran**
### Related Issues
- Issue `#1` or (None)

## Checklist
- [x] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

## Related PRs
- Related PR `#1` or (None)

## Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/secrets/api/secret-type.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.POST,
+        url: store.getState().config.endpoints.createSecretType
+    };
+
+    try {
+        const response: AxiosResponse<SecretTypeModel> = await httpClient(requestConfig);
+        if (response.status !== HttpCodes.CREATED) {
+            return Promise.reject(new Error(`Failed to create the secret type ${ body.name }.`));
```
Comment (brionmario) [inline apps/console/src/features/secrets/api/secret-type.ts:58]: Let's throw `IdentityAppsAPIException` here.

Comment (brionmario) [reply apps/console/src/features/secrets/api/secret-type.ts:58]: Check other places as well.

File: apps/console/src/features/secrets/api/secret.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
+import { IdentityClient } from "@wso2/identity-oidc-js";
```
Comment (brionmario) [inline apps/console/src/features/secrets/api/secret.ts:1]: Missing license header.

File: apps/console/src/features/secrets/components/add-secret-wizard.tsx [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+}
+
+/**
+ * This wizard is sorely responsible for adding a new `secret` for a
+ * given `secret-type`.
+ *
+ * Known Issue TODO: https://github.com/wso2/product-is/issues/12447
+ *                   Need to add i18n strings, access-control, event-publishers
+ * @param props {AddSecretWizardProps}
+ * @constructor
```
Comment (brionmario) [inline apps/console/src/features/secrets/components/add-secret-wizard.tsx:59]: Add return type.
```suggestion
```

File: apps/console/src/features/secrets/components/edit-secret.tsx [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+import { Divider, Grid, Message } from "semantic-ui-react";
+import { AppState, ConfigReducerStateInterface } from "../../core";
+import { updateSecret } from "../api/secret";
+import { SecretModel } from "../models/secret";
+import {
+    SECRET_DESCRIPTION_LENGTH,
+    SECRET_VALUE_LENGTH,
+    secretValueValidator
+} from "../utils/secrets.validation.utils";
+
```
Comment (brionmario) [inline apps/console/src/features/secrets/components/edit-secret.tsx:35]:
```suggestion
```

File: apps/console/src/features/secrets/configs/http-codes.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export class HttpCodes {
```
Comment (brionmario) [inline apps/console/src/features/secrets/configs/http-codes.ts:19]: Lets move this to `@wso2is/core`.

File: apps/console/src/features/secrets/constants/secrets.common.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (brionmario) [inline apps/console/src/features/secrets/constants/secrets.common.ts:2]:
```suggestion
 * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/console/src/features/secrets/models/secret-type.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+ *  "description": "This is a sample secret type's description"
+ * }
+ */
+export interface SecretTypeModel {
+    id: string;
+    name: string;
+    description: string;
+}
+
+// Request DTOs
```
Comment (brionmario) [inline apps/console/src/features/secrets/models/secret-type.ts:35]: Lets avoid section comments.

**Review by brionmario — APPROVED** [review]

**Review by thivi — APPROVED** [review]

**Pull request merged**